src/components/MenuSelect.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import Template from './Template';
+import autoHideContainerHOC from '../decorators/autoHideContainer.js';
+import headerFooterHOC from '../decorators/headerFooter.js';
+
+class MenuSelect extends Component {
+  static propTypes = {
+    cssClasses: PropTypes.shape({
+      select: PropTypes.string,
+      option: PropTypes.string,
+    }),
+    items: PropTypes.array.isRequired,
+    refine: PropTypes.func.isRequired,
+    templateProps: PropTypes.object.isRequired,
+  };
+
+  get selectValue() {
+    const selectValue = this.props.items.find(({ isRefined }) => isRefined);
+    return selectValue || { value: 'all' };
+  }
+
+  handleSelectChange = ({ target: { value } }) => {
+    this.props.refine(value === 'all' ? this.selectValue.value : value);
```

Comment:
That's interesting. You implement a special value for the selection. I think it should be part of the connector API. I would go for `undefined` or `null` (or both) as `refine(null)` is semantically close to "refine nothing" which is what we want when implementing "no filter / all".

Reply:
For reference, this is what a user tried to do: https://stackoverflow.com/questions/46000184/algolia-instantsearch-connectmenu-refine-remove-current-refined-value

Reply:
https://github.com/algolia/instantsearch.js/issues/2309

Reply:
Yes, I don't really know if it's the right time, we need to introduce this change a bit everywhere (according to the issue) like on the connectRefinementList connectHieararchicalMenu etc...

We can do this in the next step?